src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -53,10 +48,17 @@ private bool IsUserLog<TState>(TState state)
                 Utility.GetStateBoolValue(stateDict, ScriptConstants.LogPropertyIsUserLogKey) == true);
         }
 
+        private bool IsDeferredLog(IDictionary<string, object> scopeProps)
+        {
+            return scopeProps.Keys.Contains(ScriptConstants.LoggerDeferredLog);
+        }
+
         public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
         {
+            IDictionary<string, object> scopeProps = DictionaryLoggerScope.GetMergedStateDictionary() ?? new Dictionary<string, object>();
+
             // User logs are not logged to system logs.
-            if (!IsEnabled(logLevel) || IsUserLog(state))
+            if (!IsEnabled(logLevel) || IsUserLog(state) || IsDeferredLog(scopeProps))
```

Comment:
You skip deferred because they will be flushed later?

Reply:
I skip deferred here b/c we have a SystemLogger at the WebHost level that logs directly (without deferring). So if we wrote them here, we'd get dupes.